### ISSUE TITLE
Dropdown does not appear with certain circumstances

### ISSUE THREAD

**Issue by sam11927** [Esri/calcite-design-system#10711]
### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Actual Behavior

Dropdowns in a panel with certain container elements will disappear alongside the action if you try to open it. This is a problem because we need to be able to make the dropdown appear in front of other elements and even other panels but instead this makes everything disappear :(

### Expected Behavior

Dropdown and action should not disappear and should be able to appear in front of other panels. 

### Reproduction Sample

https://codepen.io/sam11927/pen/BaEbOgw?editors=1011

### Reproduction Steps

click the dropdown once

### Reproduction Version

2.13

### Priority impact

impact - p1 - need for current milestone

### Impact

We want this for the current R03 release--but is tagged as low priority in our release.

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/calcite-ui-icons
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Map Viewer

**Comment by macandcheese** [comment]
Could you try `overlay-positioning="fixed"` on the Dropdown to workaround this? https://codepen.io/mac_and_cheese/pen/abeRMvr?editors=1011

**Comment by sam11927** [comment]
Unfortunately I don't see changing the overlay-positioning to either of the two options fixing this issue

**Comment by macandcheese** [comment]
It’s possible there are other conflicting containers - could you share a link to the implementation (on Teams) when you get a chance? Thanks

**Comment by macandcheese** [comment]
More context
![Screenshot 2024-11-12 at 10 06 22 AM](https://github.com/user-attachments/assets/d9ba855d-5d48-4494-98f7-7a5c10968d85)
![Screenshot 2024-11-12 at 10 07 13 AM](https://github.com/user-attachments/assets/8a269276-3c3f-4b2d-9344-418bfa3e0e8d)

**Comment by macandcheese** [comment]
Follow up from triage @jcfranco @geospatialem @driskull - We could maybe add `flipPlacement` here like other Floating UI components -  which might allow some better fallbacks here.

**Comment by geospatialem** [comment]
With the `overlayPositioning` and newly added `flipPlacements` the above _should_ be mitigated for `dropdown`. However, it looks like the original Codepen above isn't specified to the component. @sam11927 Does the [following solution](https://codepen.io/geospatialem/pen/ByyxvGm) with `flipPlacements` work for the team?

**Comment by sam11927** [comment]
Yes the solution in the codepen along with some other changes in our code is sufficient for our use case! Thank you for the help @geospatialem :)

**Comment by geospatialem** [comment]
Awesome, thank you for confirming, Samantha! Will close out the above!
